pkg/controller/autoscaling/elasticsearch/driver.go

```diff
@@ -114,6 +114,7 @@ func (r *baseReconcileAutoscaling) attemptOnlineReconciliation(
 	if err != nil {
 		return nil, err
 	}
+	defer esClient.Close()
```

Comment:
I think we would need the same here https://github.com/elastic/cloud-on-k8s/blob/fa9234eaab21d76597706d8300d153d9ea631997/pkg/controller/stackconfigpolicy/controller.go#L714